sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/RequiresDistributionAndOrdering.java

```diff
@@ -42,6 +42,15 @@
    */
   Distribution requiredDistribution();
 
+  /**
+   * Returns the number of partitions required by this write.
+   * <p>
+   * Implementations may want to override this if it requires the specific number of partitions.
+   *
+   * @return the required number of partitions, non-positive values mean no requirement.
```

Comment:
Rather than "non-positive" how about "any value less than 1"? I think that's more clear.

Reply:
Probably you've worried about the possible confusion on inclusion of "0". While non-positive isn't wrong I agree it's less intuitive (less straightforward). Thanks for the suggestion!